wagtail/users/views/users.py

```diff
@@ -180,57 +182,72 @@ def create(request):
+            update_session_auth_hash(self.request, self.object)
+        return instance
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        kwargs.update({"editing_self": self.editing_self})
+        return kwargs
+
+    def run_before_hook(self):
+        for fn in hooks.get_hooks("before_edit_user"):
+            result = fn(self.request, self.object)
+            if hasattr(result, "status_code"):
+                return result
 
-    for fn in hooks.get_hooks("before_edit_user"):
-        result = fn(request, user)
-        if hasattr(result, "status_code"):
-            return result
-    if request.method == "POST":
-        form = get_user_edit_form()(
-            request.POST, request.FILES, instance=user, editing_self=editing_self
+    def run_after_hook(self):
+        for fn in hooks.get_hooks("after_edit_user"):
+            result = fn(self.request, self.object)
+            if hasattr(result, "status_code"):
+                return result
+
+    def get_success_buttons(self):
+        return (
+            [
+                messages.button(
+                    reverse("wagtailusers_users:edit", args=(self.object.pk,)),
+                    _("Edit"),
+                )
+            ],
         )
-        if form.is_valid():
-            with transaction.atomic():
-                user = form.save()
-                log(user, "wagtail.edit")
 
-            if user == request.user and "password1" in form.changed_data:
-                # User is changing their own password; need to update their session hash
-                update_session_auth_hash(request, user)
+    def get_edit_url(self):
+        return reverse(self.edit_url_name, args=(self.object.pk,))
 
-            messages.success(
-                request,
-                _("User '{0}' updated.").format(user),
-                buttons=[
-                    messages.button(
-                        reverse("wagtailusers_users:edit", args=(user.pk,)), _("Edit")
-                    )
-                ],
-            )
-            for fn in hooks.get_hooks("after_edit_user"):
-                result = fn(request, user)
-                if hasattr(result, "status_code"):
-                    return result
-            return redirect("wagtailusers_users:index")
-        else:
-            messages.error(request, _("The user could not be saved due to errors."))
-    else:
-        form = get_user_edit_form()(instance=user, editing_self=editing_self)
+    def get_delete_url(self):
+        return reverse(self.delete_url_name, args=(self.object.pk,))
 
-    return TemplateResponse(
-        request,
-        "wagtailusers/users/edit.html",
-        {
-            "user": user,
-            "form": form,
-            "can_delete": can_delete,
-        },
-    )
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data()
+        context["can_delete"] = self.can_delete
+        return context
```

Comment:
```suggestion
        context.pop('action_url')
        return context
```

We are getting an empty add button - a note to set this add_url to None to avoid this.

<img width="1254" alt="Screen Shot 2022-10-04 at 3 28 21 pm" src="https://user-images.githubusercontent.com/1396140/193741560-23db5cbd-3470-403e-914b-cd7133f5dec6.png">